site/src/pages/api/completeSignup.api.ts

```diff
@@ -0,0 +1,54 @@
+import { body as bodyValidator, validationResult } from "express-validator";
+import {
+  createAuthenticatedHandler,
+  AuthenticatedApiRequest,
+} from "../../lib/handler/authenticatedHandler";
+import { BaseApiResponse } from "../../lib/handler/baseHandler";
+import { SerializedUser } from "../../lib/model/user.model";
+import { formatErrors } from "../../util/api";
+
+export type ApiPutMeRequestBody = {
+  shortname: string;
+  preferredName: string;
+};
+
+export type ApiMeResponse = {
+  user: SerializedUser;
+};
+
+export default createAuthenticatedHandler<undefined, ApiMeResponse>(false)
+  .use(
+    bodyValidator("shortname").isString().notEmpty(),
+    bodyValidator("preferredName").isString().notEmpty(),
+  )
+  .post(
+    async (
+      req: AuthenticatedApiRequest & { body: ApiPutMeRequestBody },
+      res: BaseApiResponse<ApiMeResponse>,
+    ) => {
+      const errors = validationResult(req);
+      if (!errors.isEmpty()) {
+        return res.status(400).json(formatErrors(...errors.array()));
+      }
+
+      const { db, user, body } = req;
+
+      const { shortname, preferredName } = body;
+
+      if (user.shortname) {
```

Comment:
you could only throw if they are trying to set a different one

```suggestion
      if (user.shortname !== shortname) {
```

Reply:
Sure 👍